File: moban_java/src/main/java/org/jdk/project/controller/MonitorController.java
package org.jdk.project.controller;

import lombok.RequiredArgsConstructor;
import org.jdk.project.dto.ListResponse;
import org.jooq.generated.project.tables.pojos.LoginLog;
import org.jooq.generated.project.tables.pojos.OperationLog;
import org.jdk.project.service.MonitorService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/monitor")
@RequiredArgsConstructor
public class MonitorController {

  private final MonitorService monitorService;

  @GetMapping("/login-logs")
  public ListResponse<LoginLog> listLoginLogs() {
    return monitorService.listLoginLogs();
  }

  @GetMapping("/operation-logs")
  public ListResponse<OperationLog> listOperationLogs() {
    return monitorService.listOperationLogs();
  }
}

File: backend/src/main/kotlin/org/jdk/project/controller/MonitorController.kt
package org.jdk.project.controller

import org.jdk.project.dto.ListResponse
import org.jdk.project.dto.log.LoginLogDto
import org.jdk.project.dto.log.OperationLogDto
import org.jdk.project.service.LogService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/monitor")
class MonitorController(
    private val logService: LogService,
) {
    @GetMapping("/login-logs")
    fun listLoginLogs(): ListResponse<LoginLogDto> = logService.listLoginLogs()

    @GetMapping("/operation-logs")
    fun listOperationLogs(): ListResponse<OperationLogDto> = logService.listOperationLogs()

    @DeleteMapping("/login-logs")
    fun clearLoginLogs() {
        logService.clearLoginLogs()
    }

    @DeleteMapping("/operation-logs")
    fun clearOperationLogs() {
        logService.clearOperationLogs()
    }
}
